elastic/kibana
# [Fleet] Support integration secrets with `required: false`

## Summary

Support secrets with `required: false` in package manifests.

Closes #172061

## To test

1. Set up an integration in a local package registry with a variable that has `secret: true` and `required: false`, e.g.

```yml
- name: secret_token
  type: password
  title: (Test) Secret Token
  description: |
    Test non-required secret
  show_user: true
  secret: true
  required: false
```

2. Create a package policy for your test package and note the optional secret is rendered properly
3. Submit the policy editor form without filling out a value for the optional secret
4. Observe the request is successful
5. Edit the package policy and set a value for the optional secret
6. Observe that the secret creation logic works as expected

## Screen recording

https://github.com/elastic/kibana/assets/6766512/36e271c5-29d0-49f8-91e8-abc6a7871b20
